pkgs/by-name/st/stoat-desktop/package.nix

```
@@ -0,0 +1,82 @@
+{
+  stdenvNoCC,
+  lib,
+  asar,
+  fetchurl,
+  fetchzip,
+  makeWrapper,
+  electron,
+}:
+
+stdenvNoCC.mkDerivation rec {
+  pname = "stoat-desktop";
+  version = "1.1.11";
+
+  src = fetchzip {
+    url = "https://github.com/stoatchat/for-desktop/releases/download/v${version}/Stoat-linux-x64-${version}.zip";
+    hash = "sha256-ZVITdGwi5gh3AEttwvWQJwBQrAFqYj//Z8C+qFWuGQ0=";
+    stripRoot = false;
+  };
+
+  dontConfigure = true;
+  dontBuild = true;
+
+  nativeBuildInputs = [
+    makeWrapper
+    asar
+  ];
+
+  installPhase = ''
+    runHook preInstall
+    set -euo pipefail
+
+    mkdir -p "$out/bin" "$out/share/applications" "$out/share/stoat-desktop"
+
+    asarPath="$(find "$src" -type f -path '*/resources/app.asar' -print -quit || true)"
+    if [ -z "$asarPath" ]; then
+      echo "ERROR: could not find resources/app.asar in $src" >&2
+      exit 1
+    fi
+
+    resDir="$(dirname "$asarPath")"
+    appDir="$(dirname "$resDir")"
+
+    cp -a "$resDir" "$out/share/stoat-desktop/resources"
+    if [ -d "$appDir/locales" ]; then
+      cp -a "$appDir/locales" "$out/share/stoat-desktop/locales"
+    fi
+
+    mkdir -p "$out/share/icons/hicolor/scalable/apps"
+    install -m644 ${
+      fetchurl {
+        url = "https://raw.githubusercontent.com/stoatchat/assets/628eb2f8255c215148007f4227d2e0d5d1a67582/desktop/icon.svg";
+        hash = "sha256-EMDwnAtfTZqhw8hMGTbMjZUrnVkP+2HkyHlSy6pZmik=";
+      }
+    } "$out/share/icons/hicolor/scalable/apps/stoat-desktop.svg"
+
+    substitute ${./stoat-desktop.desktop} "$out/share/applications/${pname}.desktop" --subst-var out
```

Substituting pname like this is an anti pattern.

```suggestion
    substitute ${./stoat-desktop.desktop} "$out/share/applications/stoat-desktop.desktop" --subst-var out
```